aips/aip-32.md

```diff
@@ -48,38 +48,29 @@ No visible change to how one writes Move. But the economics changes:
-- `storage_refund_degrade_start` (microseconds) If a slot is deleted when younger than this, the maximum ratio is used for refund.
-- `storage_refund_degrade_period` (microseconds) Refund degrading period. Within the period the refund degrades linearly from the max ratio to the min ratio defined above;  after the period the min refund ratio is used.
-
-![deletion_refund_ratio](../diagrams/deletion_refund_ratio.jpeg)
+- The current on-chain timestamp and the amount paid for slot allocation will be tracked as metadata attached to the slot. For simplicity, the refundable part of the gas charge will be burnt even if other parts of the gas charge are redistributed, lowering the global supply of the native token.
+- For a deleted slot, a refund is mintted and issued to the transaction payer, increasing the global supply of the native token.
 
 # Reference Implementation
 
-https://github.com/aptos-labs/aptos-core/pull/6514
+[TBD]
 
 # Risks and Drawbacks
 
-- Cost from refunding to multiple accounts: a transaction releasing a lot of slots can potentially involve a lot of storage writes; this is protected by ensuring those writes are charged according to the normal gas schedule and write set size limits.
+1. Refunding to the deleter makes it profitable for a malicious module developer to update the code and delete user data.
+2. Refunding to the deleter makes it easy to resell storage slots, which promotes speculation on storage pricing.
+3. Full refund provides no protection against reserving storage for future usage or reselling. In expectation of higher storage pricing, one can speculatively reserve storage slots for long at virtually no cost (aside from locking up the value).
 
 # Future Potential
 
-## Billing accounts
+## Declining Refund Ratio Over Time
+
+To protect against speculation on storage pricing and incentivize timely storage clean up, the refund can be implemented as declining over time -- If deleted swiftly, the refund will be full, and over time the refund ratio declines to a configurable minimal.
 
-Extra semantics can be implemented so that a party other than the transaction sender can pay for selective storage allocations involved in the transaction. This allows the contract developer to pay for shared storage slots. Because deletions are refunded, maintaining a small active set of shared slots, despite allocations and deallocations happen frequently, imposes minimal cost but smoothens user experience. It also enables various economic activities like subsidizing the cost between different contracts.
+## Storage Deposit
 
-This is worth a separate AIP as a follow up.
+The refundable amount can be tracked globally or on a per acount basis (call it storage deposit), providing clarity and potentially enables incentivizing the validators for storing the onchain data by generating and redistributing interest from the deposit.
```

Comment:
considering that this is really easy to do with an indexer and doesn't impact the fullnode itself, is there any reason for this?

I don't follow the incentivization model.